File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

vector<string> solution(vector<string> players, vector<string> callings) {
    vector<string> answer;
    map<string, int> m;
	for (int i = 1; i <= players.size(); i++) {
		m.insert({ players[i - 1], i });
	}
	for (int i = 0; i < callings.size(); i++) {
		string temp = "";
		if (m.find(callings[i]) != m.end()) {
			temp = players[m.find(callings[i])->second - 1];
			players[m.find(callings[i])->second - 1] = players[m.find(callings[i])->second - 2];
			players[m.find(callings[i])->second - 2] = temp;
            
            m[callings[i]]--;
            m[players[m.find(callings[i])->second]]++;
		}
	}
    return players;
}

File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.cs
using System;
using System.Collections.Generic;

public class Solution {
    public string[] solution(string[] players, string[] callings) {
        Dictionary<string, int> dic = new Dictionary<string, int>();    
        for (int i = 1; i <= players.Length; i++)
        {
            dic.Add(players[i - 1], i);
        }
        for (int i = 0; i < callings.Length; i++) {
            if (dic.ContainsKey(callings[i]))
            {
                int index = dic[callings[i]];
                string temp = players[index - 1];
                players[index - 1] = players[index - 2];
                players[index - 2] = temp;

                dic[callings[i]]--;
                dic[players[index - 1]]++;
            }
        }
        return players;
    }
}